### PR TITLE
vagrant: Ported Vagrantfile from MVP.

### PR DESCRIPTION
**Pull request by ashcrow**

### CODE REVIEW

**Comment by ashcrow** [comment]
[(3) [multi-host-mgr-2] Port vagrant file](https://trello.com/c/vjXhv8Az/531-3-multi-host-mgr-2-port-vagrant-file)

File: doc/development.rst [hunk 1]
```diff
@@ -29,30 +29,43 @@ Vagrant
...
 .. todo::
 
-    Currently not ported to new architecture.
+A ``Vagrantfile`` is provided which will give you a full local development setup. 
+
+To run the vagrant development environment make sure you have a supported
+virtualization system, vagrant installed, and have all commissaire project checked
+out in the parent folder as the commissaire vagrant box will attempt to mount them
+over NFS.
+
```
Comment (gbraad) [inline doc/development.rst:36]: You have dropped the actual `vagrant up` command.

Comment (ashcrow) [reply doc/development.rst:36]: Added it back in. Thanks!

File: doc/development.rst [hunk 2]
```diff
@@ -27,32 +27,46 @@ your system with application only dependencies while you code.
 Vagrant
 -------
 
-.. todo::
+A ``Vagrantfile`` is provided which will give you a full local development setup.
 
-    Currently not ported to new architecture.
+To run the vagrant development environment make sure you have a supported
+virtualization system, vagrant installed, and have all commissaire project checked
```
Comment (mbarnes) [inline doc/development.rst]: "Project" should be plural here, right?  And not sure if "commissaire-related projects" would be clearer?

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    # Development Kubernetes server.
+    # NOTE: This must start after etcd.
+    config.vm.define "kubernetes", autostart: false do |kubernetes|
+      kubernetes.vm.box = "fedora/24-cloud-base"
+      kubernetes.vm.network "private_network", ip: "192.168.152.102"
+      kubernetes.vm.provision "shell", inline: <<-SHELL
+        echo "==> Setting hostname"
+        sudo hostnamectl set-hostname kubernetes
+        echo "===> Updating the system"
+        #sudo dnf update --setopt=tsflags=nodocs -y
```
Comment (mbarnes) [inline Vagrantfile]: Did you intend to leave this commented out?  If so, maybe just remove it then?

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    commissaire.vm.network "private_network", ip: "192.168.152.100"
+    config.vm.synced_folder ".", "/vagrant", disabled: true
+    config.vm.synced_folder ".", "/vagrant/commissaire",  mount_options: ['vers=3']
+    config.vm.synced_folder "../commissaire-http", "/vagrant/commissaire-http",  mount_options: ['vers=3']
+    config.vm.synced_folder "../commissaire-service", "/vagrant/commissaire-service",  mount_options: ['vers=3']
+    commissaire.vm.provision "shell", inline: <<-SHELL
+      echo "==> Setting hostname"
+      sudo hostnamectl set-hostname commissaire
+      echo "===> Updating the system"
+      #sudo dnf update -y
```
Comment (mbarnes) [inline Vagrantfile]: Ditto above.  Was this intentional?

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      sudo cp /vagrant/commissaire-http/conf/systemd/commissaire-server.service /etc/systemd/system/commissaire-server.service
+      sudo chmod 644 /etc/systemd/system/commissaire-server.service
+      sudo mkdir --parents /etc/commissaire
+      sudo cp /vagrant/commissaire-http/conf/commissaire.conf /etc/commissaire/commissaire.conf
+      sudo sed -i 's|"listen-interface": "127.0.0.1"|"listen-interface": "192.168.152.100"|g' /etc/commissaire/commissaire.conf
+      sudo sed -i 's|"bus-uri": "redis://127.0.0.1:6379/"|"bus-uri": "redis://192.168.152.101:6379/"|g' /etc/commissaire/commissaire.conf
+      sudo sed -i 's|^ExecStart=.*|ExecStart=/bin/bash -c ". /home/vagrant/commissaire_env/bin/activate \\&\\& commissaire-server -c /etc/commissaire/commissaire.conf"|' /etc/systemd/system/commissaire-server.service
+      sudo sed -i 's|Type=simple|\&\\nWorkingDirectory=/vagrant|' /etc/systemd/system/commissaire-server.service
+      echo "===> Starting commissaire-server"
+      sudo systemctl daemon-reload
```
Comment (mbarnes) [inline Vagrantfile]: Could the `daemon-reload` be done just once at the very end after all the config tweaks are done, or do the services have to come up in sequence?  I guess I've never tested that.  Might shave a second or two off if we can get away with it.

**Review by mbarnes — CHANGES_REQUESTED** [review]

File: doc/development.rst (hunk 2 above)
Comment (ashcrow) [reply doc/development.rst]: Yes it should.

File: Vagrantfile (hunk 3 above)
Comment (ashcrow) [reply Vagrantfile]: I had commented it out while testing as it was taking a long time to update. Will fix.

File: Vagrantfile (hunk 4 above)
Comment (ashcrow) [reply Vagrantfile]: Nope. Will fix.

File: Vagrantfile (hunk 5 above)
Comment (ashcrow) [reply Vagrantfile]: Reorganized in the fixup.

**Comment by ashcrow** [comment]
:arrow_up_small:

**Review by mbarnes — APPROVED** [review]

**Pull request merged**